web/src/pages/Dashboard.tsx

```diff
@@ -0,0 +1,241 @@
+import { IngestChart, useIngestChartData } from "@/components/IngestChart";
+import { PixiMap } from "@/components/PixiMap";
+import { SetupDatabaseButton } from "@/components/SetupDatabaseButton";
+import { Stats } from "@/components/Stats";
+import { selectableCitiesData } from "@/data/constants";
+import { useUpdateCityList } from "@/data/models/useUpdateCityList";
+import { City } from "@/data/queries";
+import {
+  connectionConfig,
+  isUpdatingCities,
+  selectedCities as selectedCitiesFromRecoil,
+  selectedCity,
+  simulatorEnabled,
+} from "@/data/recoil";
+import { useNotificationsRenderer } from "@/render/useNotificationsRenderer";
+import { useConnectionState } from "@/view/hooks/hooks";
+import { useSimulationMonitor } from "@/view/hooks/useSimulationMonitor";
+import { useSimulator } from "@/view/hooks/useSimulator";
+
+const StatsWrapper = () => {
+  const config = useRecoilValue(connectionConfig);
+  const ingestData = useIngestChartData(
+    config,
+    "locations",
+    "requests",
+    "purchases",
+    "notifications",
+    "subscriber_segments"
+  );
+
+  const [selectedCities] = useRecoilState(selectedCitiesFromRecoil);
+  const [isUpdating] = useRecoilState(isUpdatingCities);
+  const { onCreateCity, onRemoveCity } = useUpdateCityList(config);
+  const [lastSelectedCityId, setLastSelectedCityId] =
+    useRecoilState(selectedCity);
+  const { colorMode } = useColorMode();
+  const [totalSelectableCities, setTotalSelectableCities] =
+    React.useState(selectedCities);
+
+    const getNewSelectedCityAfterDeletion = (city: City): number => {
+      const cityIndex = selectedCities.findIndex(c => c.id === city.id);
+      if(cityIndex === 0) {
+        if(selectedCities.length > 1) {
+          return selectableCitiesData[1].id;
+        }
+      }
+      return -1;
+    };
+
+  const getCheckedFontColor = (city: City) => {
+    if (selectedCities.map((c) => c.id).includes(city.id)) {
+      if (colorMode === "light") {
+        return "purple.500";
+      } else {
+        return "purple.200";
+      }
+    }
+    return undefined;
+  };
+
+  const setCheckItem = (city: City, checkStatus: boolean) => {
+    if (checkStatus) {
+      onCreateCity(city.centerLat, city.centerLon);
+      setLastSelectedCityId(city.id);
+    } else {
+      const noOfSelectedCities = selectedCities.length;
+      if (noOfSelectedCities <= 0) {
+        setLastSelectedCityId(-1);
+      } else {
+        setLastSelectedCityId(getNewSelectedCityAfterDeletion(city));
+      }
+      onRemoveCity(city.id);
+    }
+  };
+
+  React.useEffect(() => {
+    const selectableCityIds = selectableCitiesData.map((c) => c.id);
+    const unknownSelectedCities = selectedCities.filter(
+      (c) => !selectableCityIds.includes(c.id)
+    );
+    setTotalSelectableCities([
+      ...selectableCitiesData,
+      ...unknownSelectedCities,
+    ]);
+    if (lastSelectedCityId === -1 && selectedCities.length) {
```

Comment:
nit: Make length checks more explicit

```suggestion
    if (lastSelectedCityId === -1 && selectedCities.length > 0) {
```

Reply:
@sandeshvijayraj :point_up: